src/services/PodManager.ts

```diff
@@ -106,22 +107,27 @@ export default class PodManager {
       }
     }
 
-    await this.pods.addPod({
-      id: podSpec.id,
-      running: true,
-      duration,
-      memory: checkMemory(podSpec.resource)
-    })
+    try {
+      await this.hyperClient.runPod(podSpec)
```

Comment:
If the codius instance doesn't yet know about the pod, couldn't the pod run into issues when it performs operations at startup? Imagine if it wants to query its own metadata, but the host doesn't have an entry for it yet. That could lead to some unexpected behavior

Reply:
is that why the add pod to the leveldb needs to be before the pod being run? 
I didn't see a reason for it but if it can access that data on startup then that needs to be first and we can clean up if it fails.

Reply:
I think that might be a better approach. But then we need to ensure that two concurrent requests where the first one succeeds and the second one fails doesn't cause the contract to be deleted

Reply:
ahh, great point! Lets not hold up 1.2.0 for this. We need to address that concurrent requests issue.

Reply:
Ok, that sounds good

Reply:
Take a look at this at now, this is the new approach to verify running pods with the hyper list api being checked against our leveldb.